tests/unit/helpers/RemoteControlTest.php

```diff
@@ -589,4 +588,138 @@ public function testGetFieldmap()
     }
+  
+    /**
+     * Test the get_uploaded_files API call using response ID.
+     */
+    public function testGetUploadedFilesByResponseId()
+    {
+        $uploadedFileName = 'fu_yshu88ibxznfvbu'; // Taken from the LSA
+        $this->prepareTestWithUploadedFile('survey_archive_getFileUploadTest.lsa', 'dalahorse.jpg', $uploadedFileName);
+
+        // Create handler.
+        $admin   = new \AdminController('dummyid');
+        $handler = new \remotecontrol_handle($admin);
+
+        // Get session key.
+        $sessionKey = $handler->get_session_key(
+            self::$username,
+            self::$password);
+        $this->assertNotEquals(['status' => 'Invalid user name or password'], $sessionKey);
+
+        // Retrieve uploaded files
+        $responseId = 2;
+        $result = $handler->get_uploaded_files($sessionKey, self::$surveyId, null, $responseId);
+        $this->assertArrayHasKey($uploadedFileName, $result, '$result = ' . json_encode($result));
+
+        // Cleanup
+        self::$testSurvey->delete();
+        self::$testSurvey = null;
+    }
+
+    /**
+     * Test the get_uploaded_files API call using tokens
+     */
+    public function testGetUploadedFilesByToken()
+    {
+        $uploadedFileName = 'fu_xtdezsfty3v5vcm'; // Taken from the LSA
+        $this->prepareTestWithUploadedFile('survey_archive_getFileUploadTestClosed.lsa', 'dalahorse.jpg', $uploadedFileName);
+
+        // Create handler.
+        $admin   = new \AdminController('dummyid');
+        $handler = new \remotecontrol_handle($admin);
+
+        // Get session key.
+        $sessionKey = $handler->get_session_key(
+            self::$username,
+            self::$password);
+        $this->assertNotEquals(['status' => 'Invalid user name or password'], $sessionKey);
+
+        // Retrieve uploaded files
+        $token = '123456';
+        $result = $handler->get_uploaded_files($sessionKey, self::$surveyId, $token);
+        $this->assertArrayHasKey($uploadedFileName, $result, '$result = ' . json_encode($result));
+
+        // Cleanup
+        self::$testSurvey->delete();
+        self::$testSurvey = null;
+    }
+
+    /**
+     * Test the get_uploaded_files API call using tokens
+     */
+    public function testGetUploadedFilesByResponseIdAndToken()
+    {
+        $uploadedFileName = 'fu_xtdezsfty3v5vcm'; // Taken from the LSA
+        $this->prepareTestWithUploadedFile('survey_archive_getFileUploadTestClosed.lsa', 'dalahorse.jpg', $uploadedFileName);
+
+        // Create handler.
+        $admin   = new \AdminController('dummyid');
+        $handler = new \remotecontrol_handle($admin);
+
+        // Get session key.
+        $sessionKey = $handler->get_session_key(
+            self::$username,
+            self::$password);
+        $this->assertNotEquals(['status' => 'Invalid user name or password'], $sessionKey);
+
+        // Retrieve uploaded files
+        $token = '123456';
```

Comment:
We mist not check error status too ?

If we add test : one with good value + one with bad value. We need to check error too.

Reply:
Added more tests. Separate them in a new file.